pyproject.toml

```
@@ -0,0 +1,72 @@
+[project]
+name = "log-archival-bench"
+version = "0.0.1"
+description = "Benchmark tests for CLP and other log management tools."
+readme = "README.md"
+authors = [
+    { name = "YScope Inc.", email = "dev@yscope.com" }
+]
+requires-python = ">=3.9"
+
+[project.scripts]
+log-archival-bench = "log_archival_bench:main"
+
+[build-system]
+requires = ["hatchling"]
+build-backend = "hatchling.build"
+
+[dependency-groups]
+dev = [
+    "mypy>=1.18.2",
+    "openpyxl>=3.1.5",
+    "pandas>=2.3.2",
+    "pyyaml>=6.0.2",
+    "ruff>=0.13.1",
+]
```

we probably should move the three non-dev deps to the `dependencies` under `[project]` section
```suggestion
[dependency-groups]
dev = [
    "mypy>=1.18.2",
    "ruff>=0.13.1",
]
```